components/compass-runtime-agent/internal/compass/director/client.go

```diff
@@ -68,21 +68,58 @@ func (cc *directorClient) FetchConfiguration() ([]kymamodel.Application, error)
 	return applications, nil
 }
 
-func (cc *directorClient) SetURLsLabels(urlsCfg RuntimeURLsConfig) (graphql.Labels, error) {
-	eventsURLLabel, err := cc.setURLLabel(eventsURLLabelKey, urlsCfg.EventsURL)
+func (cc *directorClient) ReconcileLabels(urlsCfg RuntimeURLsConfig) (graphql.Labels, error) {
+	actualLabels, err := cc.getLabels()
 	if err != nil {
 		return nil, err
 	}
 
-	consoleURLLabel, err := cc.setURLLabel(consoleURLLabelKey, urlsCfg.ConsoleURL)
+	targetLabels := []struct {
```

Comment:
What do you think about changing target labels' type to a map? It will the code little shorter.

Reply:
Wonderful idea! It really made the code a lot shorter. 👍🏼

Reply:
Unfortunately, due to the map storing key-value pairs in no particular order, the tests fail randomly depending on the different possibilities of iterating over the map. 😞 Since the tests expect an exact order of GraphQL requests, the map approach unfortunately fails.

Reply:
I just made it like so:
```suggestion
		targetLabels := []struct {
		key string
		val string
	}{
		{key: eventsURLLabelKey, val: urlsCfg.EventsURL},
		{key: consoleURLLabelKey, val: urlsCfg.ConsoleURL},
	}
```
At least it looks better. 😆

Reply:
I have changed the tests a little so that they used Mockery instead of hand-coded logic. This should resolve ordering problem.